sonata/plugins/covers_lastfm.py

```diff
@@ -35,14 +35,17 @@ def on_cover_fetch(artist, album, on_save_cb, on_err_cb):
     opener.addheaders = [("User-Agent", make_user_agent())]
 
     # First, find the link to the master release of this album
-    search_url = "http://ws.audioscrobbler.com/2.0/?%s" % (
-        urllib.parse.urlencode({
-            "method": "album.getInfo",
-            "artist": artist,
-            "album": album,
-            "api_key": API_KEY,
-            "format": "json",
-        }))
+    album_not_given = album == "" or album == None
+    if album_not_given:
+        search_url = "http://ws.audioscrobbler.com/2.0/?%s" % ( urllib.parse.urlencode({ "method": "artist.getInfo", "artist":
```

Comment:
> I understand your remark but I coud not write this the way you suggest (due to my limited knowledge of Python). you mean that "not album" is the same as "album="" and album=None". I was not sure about that.

``not album``, in this case, would be equal to ``album="" or album=None``, as [the documention says](https://docs.python.org/3/library/stdtypes.html#truth-value-testing). In some cases, I do not advice in using this form, but here would be OK.

> Spaces: you mean there are tabs? I use vi (a bit oldfashoned).

I meant this : https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements
Please write ``(urllib.parse.urlencode({"method":`` instead of  ``( urllib.parse.urlencode({ "method":``, and so on. Also, limit your lines width to 80 characters.

> You understand that it sometimes is easier to duplicate code and change one branch a bit that not duplicating (although that is better ofcourse)? I am sorry: a beginning python programmer talking.

I would be interested to know a good example of where duplicated code *is* a good thing.

You can test your work by duplicating if it easier for you, but in this case, it will be much better **not** to duplicate, especially since I really had to look carefully to see what was the difference between the 2 code paths, although there's only **one** variable which is different.